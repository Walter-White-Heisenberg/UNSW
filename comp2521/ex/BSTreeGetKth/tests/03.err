==40197== Memcheck, a memory error detector
==40197== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==40197== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==40197== Command: ./testBSTreeGetKth
==40197== 
==40197== Invalid read of size 8
==40197==    at 0x1093AB: BSTreeGetKth (BSTreeGetKth.c:31)
==40197==    by 0x109256: main (testBSTreeGetKth.c:18)
==40197==  Address 0x8 is not stack'd, malloc'd or (recently) free'd
==40197== 
==40197== 
==40197== Process terminating with default action of signal 11 (SIGSEGV)
==40197==  Access not within mapped region at address 0x8
==40197==    at 0x1093AB: BSTreeGetKth (BSTreeGetKth.c:31)
==40197==    by 0x109256: main (testBSTreeGetKth.c:18)
==40197==  If you believe this happened as a result of a stack
==40197==  overflow in your program's main thread (unlikely but
==40197==  possible), you can try to increase the size of the
==40197==  main thread stack using the --main-stacksize= flag.
==40197==  The main thread stack size used in this run was 8388608.
==40197== 
==40197== HEAP SUMMARY:
==40197==     in use at exit: 120 bytes in 5 blocks
==40197==   total heap usage: 14 allocs, 9 frees, 16,775 bytes allocated
==40197== 
==40197== LEAK SUMMARY:
==40197==    definitely lost: 0 bytes in 0 blocks
==40197==    indirectly lost: 0 bytes in 0 blocks
==40197==      possibly lost: 0 bytes in 0 blocks
==40197==    still reachable: 120 bytes in 5 blocks
==40197==         suppressed: 0 bytes in 0 blocks
==40197== Rerun with --leak-check=full to see details of leaked memory
==40197== 
==40197== For counts of detected and suppressed errors, rerun with: -v
==40197== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
