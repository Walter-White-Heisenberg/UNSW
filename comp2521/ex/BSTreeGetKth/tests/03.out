Enter the preorder traversal of the BST: Tree:

2
 \
  5
   \
    8
   / \
  7   9

Enter k: For k = 0, BSTreeGetKth returned 2
Enter k: For k = 1, BSTreeGetKth returned 8
Enter k: 